### PR TITLE
Feat/api cloudinary

### LINKED ISSUE #3: Upload photos
## Value Statement
As a user, I want to upload photos that I took that help me planning the trip, so I have all information for this trip in one page.

## Description
Screenshot 2021-04-26 at 11 59 00

## Acceptance criteria
- [x] photo area shouold be clickable
- [x] add photos from device
- [x] save these pictures

## Tasks
- [x] add photo area
- [x] check api (cloudinary/filestack)
- [x] save input and store it in localStorage

## Size 
large

### PR DESCRIPTION
**Pull request by yvlange**
Now you can upload an image in AddForm.js, the image gets stored in localStorage and display in SavedTrips.js 🎉

### CODE REVIEW

File: src/components/AddForm.js [hunk 1]
```diff
@@ -6,28 +6,41 @@ import {
...
+    const formData = new FormData();
+    formData.append("file", imageUpload);
+    formData.append("upload_preset", "tyikvr8a");
+
+    axios
+      .post("https://api.cloudinary.com/v1_1/dyjecx1wm/image/upload", formData)
+      .then((response) => {
+        const imageURL = String(response.data.secure_url);
+        addTripsToLocalStorage({
+          id: `${destinationInput}`,
```
Comment (FriederikeM) [inline src/components/AddForm.js]: is it necessary to have the template literals here? you are not using them in the line below for the same variable

File: src/components/AddForm.js [hunk 2]
```diff
@@ -6,28 +6,41 @@ import {
...
+          destination: destinationInput,
+          activities: activitiesInput,
+          restaurants: restaurantsInput,
+          notes: notesInput,
+          photo: imageURL,
+        });
+        const trips = getTripsFromLocalStorage();
+        console.log(trips);
```
Comment (FriederikeM) [inline src/components/AddForm.js:39]: This can go :)

File: src/components/UploadPhoto.js [hunk 3]
```diff
@@ -1,11 +1,22 @@
 import "../styles/UploadPhoto.css";
-import { BsUpload } from "react-icons/bs";
+// import { BsUpload } from "react-icons/bs";
```
Comment (FriederikeM) [inline src/components/UploadPhoto.js:2]: if you don't need it, you can delete it

File: src/components/UploadPhoto.js [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
-function UploadPhoto() {
+function UploadPhoto({ id, name, value, onChange }) {
   return (
     <div className="upload">
-      <BsUpload className="upload__icon" />
-      <p>Upload Photos</p>
+      {/* <label for="file" className="upload__label">
+        <BsUpload className="upload__icon" />
+        Choose a Photo
+      </label> */}
```
Comment (FriederikeM) [inline src/components/UploadPhoto.js:10]: if you are not using the label, you can delete it

File: src/styles/UploadPhoto.css [hunk 5]
```diff
@@ -19,3 +19,11 @@
   height: 30px;
   width: 40px;
 }
+
+/* .upload__input {
+  display: none;
+} */
```
Comment (FriederikeM) [inline src/styles/UploadPhoto.css:25]: same here

**Review by FriederikeM — APPROVED** [review]
Looks good, just left a few little comments

File: src/components/AddForm.js [hunk 6]
```diff
@@ -6,28 +6,41 @@ import {
...
-    console.log(trips);
-    setDestinationInput("");
-    setActivitiesInput("");
-    setRestaurantsInput("");
-    setNotesInput("");
+    const formData = new FormData();
+    formData.append("file", imageUpload);
+    formData.append("upload_preset", "tyikvr8a");
+
+    axios
```
Comment (namirsab) [inline src/components/AddForm.js]: You don't need to use axios here. You could just use `fetch` and uninstall `axios`
You can take a look how to use `fetch` to make a post request here:

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options

File: src/components/SavedTrip.js [hunk 7]
```diff
@@ -21,6 +22,13 @@ function SavedTrip() {
       <TripDetails name="activities" text={singleTrip.activities} />
       <TripDetails name="restaurants" text={singleTrip.restaurants} />
       <TripDetails name="notes" text={singleTrip.notes} />
+      <Image
+        cloudName="dyjecx1wm"
```
Comment (namirsab) [inline src/components/SavedTrip.js:26]: Just out of curiosity, where do you get the `cloudName` from?
Maybe this is something that could be an environmental variable?

**Review by namirsab — CHANGES_REQUESTED** [review]
Impressive @yvlange !
i'm super proud how you managed to integrate cloudinary!

Just a few comments, please take a look

File: src/components/AddForm.js (hunk 6 above)
Comment (namirsab) [reply src/components/AddForm.js]: Also Johan did already a post request with axios in his project, so you can ask him as well :)

File: src/components/AddForm.js (hunk 1 above)
Comment (yvlange) [reply src/components/AddForm.js]: you're absolutely right! I'll remove it :)

File: src/components/UploadPhoto.js (hunk 3 above)
Comment (yvlange) [reply src/components/UploadPhoto.js:2]: This is for the styling later haha

File: src/components/SavedTrip.js (hunk 7 above)
Comment (yvlange) [reply src/components/SavedTrip.js:26]: The cloudname is my personal cloud name I got from cloudinary.
It needs to be in there, so the image can get fetched again from this specific folder.

**Pull request merged**